### ISSUE TITLE
Use safer test for Location in argument processor

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#109106]
In https://github.com/microsoft/vscode/pull/109049, we tried to pass an argument to a command that has properties named `uri` and `range`, and got caught by this line: https://github.com/microsoft/vscode/blob/8c91bd827848f1710d055c39d80d04c59bb55463/src/vs/workbench/api/common/extHostCommands.ts#L69 trying to convert it to a location and deleting all other properties. Is there any way to make this check safer, like can it fail if there are other properties besides the expected ones?

**Comment by jrieken** [comment]
Unfortunate... We don't do instance-of checks because TypeScript doesn't enforce constructor-usage for classes with public members only and because of that we often see folks use object-literals instead of instances. What is your expected outcome? Should those argument be massages or simply passed through?

**Comment by roblourens** [comment]
In this case we were expecting uri and range to be massaged the same way, since they did represent the same things as the same props in Location. But I think those types will be caught by some other check anyway?

**Comment by jrieken** [comment]
Yeah, the massage should happen recursively... Still, this is really tricky and didn't understand it till now. This isn't the direction extension-to-renderer but renderer-to-extension. The only thing I can image is to add some marker-property onto `modes.Location` to know that it is actually a location.
